src/server.cpp

```diff
@@ -1101,15 +1102,16 @@ void Server::Receive(float timeout)
 		pkt.clear();
 		peer_id = 0;
 		try {
+			// Round up since the target step length is the minimum step length,
+			// we only have millisecond precision and we don't want to busy-wait by calling
+			// ReceiveTimeoutMs(.., 0) repeatedly.
 			if (!m_con->ReceiveTimeoutMs(&pkt,
-					(u32)remaining_time_us() / 1000)) {
+					((u32)remaining_time_us() + 999) / 1000)) {
 				// No incoming data.
-				// Already break if there's 1ms left, as ReceiveTimeoutMs is too coarse
-				// and a faster server-step is better than busy waiting.
-				if (remaining_time_us() < 1000.0f)
-					break;
-				else
+				if (remaining_time_us() > 0.0f)
 					continue;
```

Comment:
I guess this condition will never trigger except if `ReceiveTimeoutMs` doesn't wait as long as it should, since we're already waiting >= `remaining_time_us()` with `ReceiveTimeoutMs` above.

Reply:
`Semaphore::wait()` doesn't try again on EINTR, so it can trigger.

(Sidemark: `ReceiveTimeoutMs` can also wait infinity long, looks like a problem to me, but not for this PR.)